frontend/src/components/navigation/Navigation.tsx

```diff
@@ -0,0 +1,87 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { useLocalization } from "@fluent/react";
+import styles from "./Navigation.module.scss";
+import { useIsLoggedIn } from "../../hooks/session";
+import { SignUpButton } from "./SignUpButton";
+import { SignInButton } from "./SignInButton";
+import { useProfiles } from "../../hooks/api/profile";
+import { UpgradeButton } from "./UpgradeButton";
+import { WhatsNewMenu } from "../layout/whatsnew/WhatsNewMenu";
+import { UserMenu } from "./UserMenu";
+import { AppPicker } from "./AppPicker";
+import { MenuToggle } from "./MenuToggle";
+
+/** Switch between the different pages of the Relay website. */
+export const Navigation = ({ ...props }) => {
+  const { l10n } = useLocalization();
+  const router = useRouter();
+  const isLoggedIn = useIsLoggedIn();
+  const profiles = useProfiles();
+  const homePath = isLoggedIn ? "/accounts/profile" : "/";
+  const hasPremium: boolean = profiles.data?.[0].has_premium || false;
+  const { theme, mobileMenuState, handleToggle } = props;
+
+  const ToggleButton = () => (
+    <a
+      href="#"
+      className={styles["menu-toggle"]}
+      role="menuitem"
```

Comment:
I think one of the failing tests is an accessibility check that warns that menuitems should be inside a menu, rather than its trigger. (If you get to adopt react-aria, that will take care of that automatically.)

I'm not sure what exactly the right markup here is to make this properly recognisable as a menu, but I do know that generally, the guidance is that no ARIA attributes are better than the wrong attributes, so if you're unsure either, I think the best course of action is probably to just remove them.

Reply:
removed - hope react-aria work will take care of this